tikv/gc.go

```diff
@@ -94,72 +95,124 @@ func (s *KVStore) resolveLocks(ctx context.Context, safePoint uint64, concurrenc
 	return nil
 }
 
-// We don't want gc to sweep out the cached info belong to other processes, like coprocessor.
-const gcScanLockLimit = txnlock.ResolvedCacheSize / 2
+type BaseLockResolver struct {
+	identifier string
+	store      Storage
+}
+
+func NewBaseLockResolver(identifier string, store Storage) *BaseLockResolver {
+	return &BaseLockResolver{
+		identifier: identifier,
+		store:      store,
+	}
+}
+
+func (l *BaseLockResolver) Identifier() string {
+	return l.identifier
+}
+
+func (l *BaseLockResolver) ResolveLocks(bo *Backoffer, locks []*txnlock.Lock, loc *locate.KeyLocation) (*locate.KeyLocation, error) {
+	return batchResolveLocksInARegion(bo, l.GetStore(), locks, loc)
```

Comment:
The function `batchResolveLocksInARegion` only resolves locks in the single region specified by `loc`. If error happens, it may reload the region info, but still only handles the first region containing the key `locks[0]`. If `locks` locates in more than one region, the remaining will be ignored. This works in the GC logic (or the scanlocks-resolvelocks for range logic) because it guarantees `locks` passed to this function is sorted, and the returned `KeyLocation` indicates the next key to scan the next batch, so no lock will be missed. However if you public this function to be accessible everywhere, it's very weird that it doesn't promise to completely handle the given locks and returns a `KeyLocation`, and is very very very likely to cause misusing. Please consider adjust the implementation to suite the interface's semantics or any other way to avoid the problem.